models/roles/faculty_member.py

```diff
@@ -24,3 +32,11 @@ def __str__(self):
         designation = self.get_designation_display()
         department = self.department
         return f'{person} - {designation}, {department}'
+
+    def save(self, *args, **kwargs):
+        """
+        Override save method of FacultyMember class to automatically run validators of a model.
+        """
+
+        self.full_clean()
```

Comment:
Make this validation opt-in using a keyword argument `run_validations`. Always running validations will lead to us not being able to make weird edge-case changes via the shell.

See https://github.com/IMGIITRoorkee/omniport-shell/pull/7#pullrequestreview-279956141 for example.